app/uk/gov/hmrc/cgtpropertydisposals/repositories/enrolments/TaxEnrolmentRepository.scala

```diff
@@ -43,35 +41,30 @@ trait TaxEnrolmentRepository {
 }
 
 @Singleton
-class DefaultTaxEnrolmentRepository @Inject() (mongo: ReactiveMongoComponent)(implicit
+class DefaultTaxEnrolmentRepository @Inject() (mongo: MongoComponent)(implicit
   ec: ExecutionContext
-) extends ReactiveRepository[TaxEnrolmentRequest, BSONObjectID](
+) extends PlayMongoRepository[TaxEnrolmentRequest](
+      mongoComponent = mongo,
       collectionName = "tax-enrolment-requests",
-      mongo = mongo.mongoConnector.db,
-      TaxEnrolmentRequest.format,
-      ReactiveMongoFormats.objectIdFormats
+      domainFormat = TaxEnrolmentRequest.format,
+      indexes = Seq(IndexModel(ascending("ggCredId"), IndexOptions().name("ggCredIdIndex"))),
+      extraCodecs = Seq(Codecs.playFormatCodec[TaxEnrolmentRequest](TaxEnrolmentRequest.format))
     )
     with TaxEnrolmentRepository {
 
-  override def indexes: Seq[Index] =
-    Seq(
-      Index(
-        key = Seq("ggCredId" → IndexType.Ascending),
-        name = Some("ggCredIdIndex")
-      )
-    )
-
   override def save(cgtEnrolmentRequest: TaxEnrolmentRequest): EitherT[Future, Error, Unit] =
     EitherT[Future, Error, Unit](
       preservingMdc {
-        insert(cgtEnrolmentRequest)
-          .map[Either[Error, Unit]] { result: WriteResult =>
-            if (result.ok)
+        collection
+          .insertOne(cgtEnrolmentRequest)
+          .toFuture()
+          .map[Either[Error, Unit]] { result =>
+            if (result.wasAcknowledged()) {
```

Comment:
- ℹ️ This always returns true (with the default write concern). Else is no longer necessary.